test/src/org/labkey/test/tests/targetedms/TargetedMSProteinGroupingTest.java

```diff
@@ -57,4 +71,11 @@ public void testProteinGrouping()
         peptideTable = new DataRegionTable.DataRegionFinder(getDriver()).withName("Peptides").waitFor();
         checker().verifyEquals("Incorrect group for the protein", group, peptideTable.getDataAsText(0, "PeptideGroupId"));
     }
-}
+
+    private String getCurrentSequence()
+    {
+        String unformattedString = (String) executeScript("return document.getElementById(\"peptideMap\").textContent;");
+        return unformattedString.replaceAll("[^A-Z]", "").trim().substring(1);
```

Comment:
This looks nice. But why the substring(1) - what extra character as you cutting away?

Reply:
The textContent is getting the whole table text content along with this "The number displayed on each peptide box is the rank" text. 
When we replace everything but capital letters T stays back along with sequence so substring to get the sequence string.

Reply:
That definitely deserves a comment. We could also tweak the HTML we generate to make it easier for you to omit the caption - your call

Reply:
Done..!

Reply:
Added the comment.